src/h5web/providers/hsds/utils.test.ts

```diff
@@ -0,0 +1,179 @@
+      };
+      expect(convertHsdsType(beFloatType)).toEqual({
+        class: HDF5TypeClass.Float,
+        size: 64,
+        endianness: 'BE',
+        unsigned: undefined,
+      });
+    });
+
+    it('should convert the base of Array type', () => {
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const arrayType: HsdsArrayType = {
+        class: HDF5TypeClass.Array,
+        base: integerType,
+        dims: [4, 5],
+      };
+      expect(convertHsdsType(arrayType)).toEqual({
+        class: HDF5TypeClass.Array,
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+          unsigned: undefined,
+        },
+        dims: [4, 5],
+      });
+    });
+
+    it('should convert the base of VLen type', () => {
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const vlenType: HsdsVLenType = {
+        class: HDF5TypeClass.VLen,
+        base: integerType,
+      };
+      expect(convertHsdsType(vlenType)).toEqual({
+        class: HDF5TypeClass.VLen,
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+          unsigned: undefined,
+        },
+      });
+    });
+
+    it('should convert the field types of Compound type', () => {
+      const floatType: HsdsBaseType = {
+        class: HDF5TypeClass.Float,
+        base: 'H5T_IEEE_F64LE',
+      };
+      const integerType: HsdsBaseType = {
+        class: HDF5TypeClass.Integer,
+        base: 'H5T_STD_I8LE',
+      };
+      const vlenType: HsdsVLenType = {
+        class: HDF5TypeClass.VLen,
+        base: integerType,
+      };
+      const compoundType: HsdsCompoundType = {
+        class: HDF5TypeClass.Compound,
+        fields: [
+          {
+            name: 'f1',
+            type: floatType,
+          },
+          {
+            name: 'f2',
+            type: vlenType,
+          },
+        ],
+      };
+      expect(convertHsdsType(compoundType)).toEqual({
+        class: HDF5TypeClass.Compound,
+        fields: [
+          {
+            name: 'f1',
+            type: {
+              class: HDF5TypeClass.Float,
+              size: 64,
+              endianness: 'LE',
+              unsigned: undefined,
+            },
+          },
+          {
+            name: 'f2',
+            type: {
+              class: HDF5TypeClass.VLen,
+              base: {
+                class: HDF5TypeClass.Integer,
+                size: 8,
+                endianness: 'LE',
+                unsigned: undefined,
+              },
+            },
+          },
+        ],
+      });
+    });
+
+    it('should throw when encountering an unknown type', () => {
+      const unknownType = { class: 'NO_CLASS' };
+      expect(() => convertHsdsType(unknownType as HsdsType)).toThrow(
+        /Unknown type/u
+      );
```

Comment:
Even if the error cannot happen according to TypeScript, we need to test it as TS has no control on what comes out of the HSDS requests.